src/core/net/ip6_address.hpp

```diff
@@ -233,6 +233,20 @@ class Prefix : public otIp6Prefix, public Clearable<Prefix>
      */
     bool operator!=(const Prefix &aOther) const { return !(*this == aOther); }
 
+    /**
+     * This method overloads operator `<` to compare two prefixes.
+     *
+     * A prefix with shorter length is considered smaller than the one with longer length. If the prefix lengths are
+     * equal, then the prefix bytes are compared directly.
+     *
+     * @param[in] aOther  The other prefix to compare against.
+     *
+     * @retval TRUE   If the prefix is smaller than @p aOther.
+     * @retval FALSE  If the prefix is not smaller than @p aOther.
+     *
+     */
+    bool operator<(const Prefix &aOther) const;
```

Comment:
Just some thoughts:

I think it is a good practice to always have one relational operator and dedrive the rest from it. For example, we may consider implement `operator==` with:
```c++
bool operator==(const Prefix &aOther) const
{
    return !( *this < aOther || aOther < *this);
}
```

But it requires two `<` operations and current implementation should faster :)

Reply:
Yes. I think having simpler `operator==()` would be better here. As you mention, definitley faster.

I did not add it for now, but we can later consider adding a CRTP style template class that adds support for other operators `>`, `<=`, `>=` implemented from `<` and `==` provided by the main class.  
We can also add/use similar model for adding `!=` from an existing `==` from the main class (it is practically a syntactic sugar optimization).